internal/app/tfsec/checks/aws047.go

```diff
@@ -0,0 +1,93 @@
+package checks
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/tfsec/tfsec/internal/app/tfsec/parser"
+	"github.com/tfsec/tfsec/internal/app/tfsec/scanner"
+)
+
+const AWSSqsPolicyWildcardActions scanner.RuleCode = "AWS047"
+const AWSSqsPolicyWildcardActionsDescription scanner.RuleSummary = "AWS SQS policy document has wildcard action statement."
+const AWSSqsPolicyWildcardActionsExplanation = `
+SQS Policy actions should always be restricted to a specific set.
+This ensures that the queue itself cannot be modified or deleted, and prevents
+possible future additions to queue actions to be implicitly allowed.
+`
+const AWSSqsPolicyWildcardActionsBadExample = `
+resource "aws_sqs_queue_policy" "test" {
+  queue_url = aws_sqs_queue.q.id
+
+  policy = <<POLICY
+{
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": "*",
+      "Action": "*"
+    }
+  ]
+}
+POLICY
+}
+`
+const AWSSqsPolicyWildcardActionsGoodExample = `
+resource "aws_sqs_queue_policy" "test" {
+  queue_url = aws_sqs_queue.q.id
+
+  policy = <<POLICY
+{
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": "*",
+      "Action": "sqs:SendMessage"
+    }
+  ]
+}
+POLICY
+}
+`
+
+func init() {
+	scanner.RegisterCheck(scanner.Check{
+		Code: AWSSqsPolicyWildcardActions,
+		Documentation: scanner.CheckDocumentation{
+			Summary:     AWSSqsPolicyWildcardActionsDescription,
+			Explanation: AWSSqsPolicyWildcardActionsExplanation,
+			BadExample:  AWSSqsPolicyWildcardActionsBadExample,
+			GoodExample: AWSSqsPolicyWildcardActionsGoodExample,
+			Links:       []string{},
```

Comment:
Can we have the links to any relevant aws docs, and the terraform ones for `aws_sqs_queue_policy` please

Reply:
Not sure what exactly is relevant from the AWS docs, I guess this one might work https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-security-best-practices.html especially the `Implement least-privilege access` paragraph?

Reply:
Yeah, I think that's a pretty good starting place for someone to read up. It's relevant